cli/tests/test_describe_command.rs

```diff
@@ -898,6 +898,187 @@ fn test_edit_cannot_be_used_with_no_edit() {
+    insta::assert_snapshot!(output, @r"
+    Message from CLI
+
+    Signed-off-by: Test User <test.user@example.com>
+    CC: alice@example.com
+    Change-Id: I6a6a69649a45c67d3e96a7e5007c110ede34dec5
+    [EOF]
+    ");
+
+    // invalid generated trailers won't appear in the description
+    let output = work_dir.run_jj([
+        "describe",
+        "--no-edit",
+        "--config",
+        r#"templates.commit_trailers='"this is an invalid trailer"'"#,
+    ]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Nothing changed.
+    [EOF]
+    ");
+
+    // it doesn't modify a commit with an empty description
+    let output = work_dir.run_jj(["new"]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Working copy  (@) now at: yostqsxw 5a8ea7e1 (empty) (no description set)
+    Parent commit (@-)      : qpvuntsm 2d3438dc (empty) Message from CLI
+    [EOF]
+    ");
+    let output = work_dir.run_jj([
+        "describe",
+        "--no-edit",
+        "--config",
+        r#"templates.commit_trailers='"CC: alice@example.com"'"#,
+    ]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Nothing changed.
+    [EOF]
+    ");
+}
+
+#[test]
+fn test_add_trailer_committer() {
+    let mut test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let edit_script = test_env.set_up_fake_editor();
+    let work_dir = test_env.work_dir("repo");
+    test_env.add_config(
+        r#"[templates]
+        commit_trailers = '''"Signed-off-by: " ++ self.committer().email()'''"#,
+    );
+
+    let output = work_dir.run_jj(["describe", "-m", "Message from CLI"]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Working copy  (@) now at: qpvuntsm 500b5e31 (empty) Message from CLI
+    Parent commit (@-)      : zzzzzzzz 00000000 (empty) (no description set)
+    [EOF]
+    ");
+
+    let output = work_dir.run_jj(["log", "--no-graph", "-r@", "-Tdescription"]);
+    insta::assert_snapshot!(output, @r"
+    Message from CLI
+
+    Signed-off-by: test.user@example.com
+    [EOF]
+    ");
+
+    // committer is properly set in the trailer
+    let output = work_dir.run_jj([
+        "describe",
+        "--no-edit",
+        "--config",
+        "user.email=foo@bar.org",
+    ]);
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Working copy  (@) now at: qpvuntsm b71fc9e2 (empty) Message from CLI
+    Parent commit (@-)      : zzzzzzzz 00000000 (empty) (no description set)
+    [EOF]
+    ");
+
+    let output = work_dir.run_jj(["log", "--no-graph", "-r@", "-Tdescription"]);
+    insta::assert_snapshot!(output, @r"
+    Message from CLI
+
+    Signed-off-by: test.user@example.com
+    Signed-off-by: foo@bar.org
+    [EOF]
+    ");
+
+    // trailer is added with the expected committer in the editor
+    std::fs::write(&edit_script, "dump editor0").unwrap();
```

Comment:
Can you add test for editor content with empty/non-empty descriptions?

It might be better to insert two blank lines if the original description was empty.

Reply:
You were right, it's better with two blank lines.
I've added a test that doesn't show that, unfortunately: the blank lines at the head of the file seem to be trimmed.
Is there something I could do to test the whole content of the file?

Reply:
We'll need to insert `-----` or something to ensure that blank lines exist.
```rust
insta::assert_snapshot!(format("-----\n{content}-----\n"), ..
```